### DISCUSSION TITLE
Postgres table schema

### DISCUSSION THREAD

**Discussion by FCMHUB** [Budibase/budibase#2528] [Help]
Hi, is there any way to use table schemas on Postgres datasource other than public?
The use of schemas is really common for Postgres db.

**Comment by mjashanks** [comment]
Hey @FCMHUB .  Help me out here - is it possible to specify the schema as part of the postgres connection? I can't see it here: https://www.npmjs.com/package/postgres (this is the nodejs-postgres connector).

If you were setting the schema to use, is the only way to do `select ... from myschema.mytable` ? Is there someway we can easily apply it to all queries, without trying to modify the query itself?

I was reading: https://www.postgresqltutorial.com/postgresql-schema - which seems to suggest we could run...

`SET search_path TO myschema;`

..before every query. Is this correct?

**Reply by dncpax** [reply]
Also trying bb for the first time and it is really awkward that after setting my connection to pgsql I can only see public schema tables, which is really a *not* recommended way of structuring your pgsql database - you should use schemas to organize your db and help your privileges setup and that's what the majority of pgsql users/managers do.
So limiting bb to the public schema really hampers using it with pgsql.
On the other hand the automatic features of bb producing screens for list/details/insert new are amazing! And would really benefit from working with any schema. 
I would propose to, as soon as possible, not limiting to public at all anywhere in bb, and later improve on that by letting users somehow navigate schemas (limiting a connection to a single schema will also cause trouble, since it's a very frequent affair to use several schemas in a single app).
You frequently see apps that list tables from a db using a dropdown of schemas so you can select which schema to browse in, while using a single connection. It's the db admin who is responsible for allowing access to schemas/tables.

**Reply by mjashanks** [reply]
thanks for your input @dncpax - noted!
